app/scripts/modules/core/src/pipeline/details/SingleExecutionDetails.tsx

```diff
@@ -38,108 +30,155 @@ export interface ISingleExecutionRouterStateChange extends IStateChange {
   toParams: ISingleExecutionStateParams;
 }
 
-export class SingleExecutionDetails extends React.Component<
-  ISingleExecutionDetailsProps,
-  ISingleExecutionDetailsState
-> {
-  private executionScheduler: IScheduler;
-  private executionLoader: Subscription;
-  private stateChangeSuccessSubscription: Subscription;
-
-  constructor(props: ISingleExecutionDetailsProps) {
-    super(props);
-
-    this.state = {
-      execution: null,
-      sortFilter: ExecutionState.filterModel.asFilterModel.sortFilter,
-      stateNotFound: false,
-    };
+export function getAndTransformExecution(id: string, app: Application) {
+  return ReactInjector.executionService.getExecution(id).then((execution) => {
+    ExecutionsTransformer.transformExecution(app, execution);
+    return execution;
+  });
+}
+
+//TODO: Limit the lineage to some reasonable number of max generations
+function traverseLineage(execution: IExecution): string[] {
+  const lineage: string[] = [];
+  if (!execution) {
+    return lineage;
+  }
+  let current = execution;
+  // Including the deepest child (topmost, aka current, execution) in the lineage lets us
+  // also cache it as part of the ancestry state (we just don't render it).
+  // This buys us snappier navigation to descendants because the entire lineage will be local.
+  lineage.unshift(current.id);
+  while (current.trigger?.parentExecution) {
+    current = current.trigger.parentExecution;
+    lineage.unshift(current.id);
   }
+  return lineage;
+}
 
-  private getExecution() {
-    const { executionService, $state } = ReactInjector;
-    const { app } = this.props;
+export function SingleExecutionDetails(props: ISingleExecutionDetailsProps) {
+  const ancestryRef = React.useRef([] as IExecution[]);
```

Comment:
I was thinking it felt more correct than trying to read `ancestry` without declaring it as a dep (which would cause it to endlessly run recomputing itself).

But you are right, I think it should all work without the `ref` at all.